src/main/kotlin/com/terraformation/backend/accelerator/SubmissionNotifier.kt

```diff
@@ -0,0 +1,68 @@
+package com.terraformation.backend.accelerator
+
+import com.terraformation.backend.accelerator.db.DeliverableStore
+import com.terraformation.backend.accelerator.event.DeliverableDocumentUploadedEvent
+import com.terraformation.backend.accelerator.event.DeliverableReadyForReviewEvent
+import com.terraformation.backend.customer.model.SystemUser
+import com.terraformation.backend.log.perClassLogger
+import jakarta.inject.Named
+import java.time.Duration
+import java.time.InstantSource
+import org.jobrunr.scheduling.JobScheduler
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.annotation.Lazy
+import org.springframework.context.event.EventListener
+
+@Named
+class SubmissionNotifier(
+    private val clock: InstantSource,
+    private val deliverableStore: DeliverableStore,
+    private val eventPublisher: ApplicationEventPublisher,
+    @Lazy private val scheduler: JobScheduler,
+    private val systemUser: SystemUser,
+) {
+  companion object {
+    /**
+     * Wait this long before notifying about new submissions to give the user time to submit
+     * additional items for a deliverable. This is the delay after the _last_ submission, that is,
+     * the user-visible behavior is that the countdown resets each time something new is submitted
+     * for a deliverable.
+     */
+    private val notificationDelay = Duration.ofMinutes(5)
+
+    private val log = perClassLogger()
+  }
+
+  /** Schedules a "ready for review" notification when a document is uploaded. */
+  @EventListener
+  fun on(event: DeliverableDocumentUploadedEvent) {
+    scheduler.schedule<SubmissionNotifier>(clock.instant().plus(notificationDelay)) {
```

Comment:
Much simpler approach 👍 
Do we need to worry about the corner case where service is restarted during this wait time?

Reply:
No worry about that. JobRunr (which is where the `scheduler` object comes from) is a persistent scheduler. This call actually just inserts into the database, and then a background thread periodically scans to see if there are any jobs whose times have come. It also takes care of guaranteeing that only one host in the cluster will run a given job.

Reply:
Nice! Thanks for the explanation.